### PR TITLE
Check if member has clearance when enabling boefje

### PR DESCRIPTION
**Pull request by Rieven**
### Changes
When enabling boefjes check if scan level of boefje is at least the trusted clearance level of the member enabling.

### Issue link
https://github.com/minvws/nl-kat-coordination/issues/1424

Closes https://github.com/minvws/nl-kat-coordination/issues/1424

### Proof
image

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified;
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [x] I have written unit tests for the changes or fixes I made.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have made corresponding changes to the documentation, if necessary.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Review by ammar92 — APPROVED** [review]

**Comment by TwistMeister** [comment]
QA check: Everything works ## Checklist for QA:
- [ ] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
* Not enabling a plugin when clearance level isn't sufficient.
* Enabling plugin when clearance level is sufficient

Suggestion:
Screenshot 2023-09-04 at 14 23 15

I would like the "L-1" in the screenshot to be rendered differently. Something along the lines of "Your clearance level has not yet been set. Contact your administrator"

File: rocky/rocky/locale/nl/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -1908,6 +1908,14 @@ msgstr ""
 msgid "Boefje '{boefje_name}' enabled."
 msgstr "Boefje '{boefje_name}' aangezet."
 
+#: katalogus/views/plugin_enable_disable.py
+msgid "Your clearance level has not yet been set. Contact your administrator."
+msgstr "Uw vrijwaring niveau is nog niet ingesteld. Neem contact op met uw beheerder."
+
+#: katalogus/views/plugin_enable_disable.py
+msgid "To enable {} you need at least a clearance level of L{}. "
+msgstr "Om {} in te schakelen heeft u minimaal een vrijwaring niveau van L{} nodig.
```
Comment (ammar92) [inline rocky/rocky/locale/nl/LC_MESSAGES/django.po:1917]: String not closed?

Comment (ammar92) [inline rocky/rocky/locale/nl/LC_MESSAGES/django.po:1917]:
```suggestion
msgstr "Om {} in te schakelen heeft u minimaal een vrijwaring niveau van L{} nodig."
```

**Review by ammar92 — COMMENTED** [review]
Looks good to me, but there's a build error, I think it has to do with this [line](https://github.com/minvws/nl-kat-coordination/pull/1690/files#diff-2fa88be331e5988e34169b5b409976ea952a6f1b389cc1e91460f2268160246cR1917) not being closed as a string properly. This is probably also why the Debian packaging build fails.

**Comment by TwistMeister** [comment]
It appears that a red teamer can enable/disable boefjes, without having to accept the assigned clearance level. That's not intended. The organization member clearance level should not be active until it's accepted by the user.

**Comment by TwistMeister** [comment]
> Trusted clearance level has been decreased from L2 --> L0. Member has not acknowledged L0. (Member holds the old L2 clearance). Here we can only check as well for acknowledged clearance, but not preferred as the admin lowered it, is more risky.

If I'm correct this isn't the case. If the level gets lowered by the admin the "acknowledged" will follow this new, lower, level automatically. So you should be save by only checking the acknowledged level.

**Review by TwistMeister — APPROVED** [review]

**Review by underdarknl — APPROVED** [review]

**Pull request merged**